### PR TITLE
Enhance Moving Allowances page

### PR DESCRIPTION
**Pull request by jgarber623-gov**
## Checklist

I have…

- [x] built and served the site locally (`bundle exec rake jekyll:serve`) and verified that my changes behave as expected.
- [ ] run the test suite (`bundle exec rake`) and verified that all tests pass.
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.

**Note:** The test suite does not currently pass owing to some `<a href="#TODO">` anchors in the content.

## Summary of Changes

This pull request allows a user to select their rank from a list and view the entitlements available to them based on the provided rank. It _does not yet_ implement search across the dependency status and move type `<select>`s. A forthcoming PR will address those shortcomings.

From a technical perspective, this PR adds [jQuery](http://jquery.com/) (version 3.2.1) to the vendor'ed assets and establishes a rough (and imperfect) pattern for developing and implementing JavaScript components. The current strategy relies on fairly uniquely-named components (`MovingAllowances`, in this case) which are methods attached to `window` and instantiated with the `new` keyword. _This is by no means a theoretically perfect module development pattern_, but it suits the needs of this site for now.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo and set up development dependencies,
1. `git checkout enhance-moving-allowances-page`,
1. `bundle exec rake jekyll:serve`,
1. point your Web browser of choice at http://localhost:4000/moving-allowances/determine-your-allowance/, and
1. observe that the site renders and behaves as shown in the screenshots below.

### Narrow viewport

narrow

### Wide viewport

wide

☝️ The yellow background is a small design enhancement to indicate that the `<table>` is in focus. @lfantano, let me know if you have opinions on this choice!

### CODE REVIEW

File: src/_assets/javascripts/components/moving-allowances.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            } else {
+              hideElement($table);
+            }
+          }.bind(this));
+        } else {
+          hideElement(this.$container);
+        }
+      }
+    }
+  };
```
Comment (jgarber623-gov) [inline src/_assets/javascripts/components/moving-allowances.js]: For a relatively simple module, this `events` sub-object is a bit over overkill, but I'm hoping to establish some patterns early on for how to organize a component's `Foo.prototype`.

File: src/_assets/stylesheets/elements/_inputs.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+select {
+  padding-right: 4rem;
+}
```
Comment (jgarber623-gov) [inline src/_assets/stylesheets/elements/_inputs.scss]: Solving this particular UI bug. 👇 

screen shot 2017-05-15 at 4 09 37 pm

File: src/_vendor/assets/stylesheets/uswds/elements/_inputs.scss [hunk 3]
```diff
@@ -9,7 +9,7 @@ $input-padding-vertical: 1rem;
 $input-height-exact: (
   ($base-font-size * $input-line-height) +
   ($input-padding-vertical * 2) +
-  (strip-units($input-border-width) * 2rem / $base-font-size)
+  (rem($input-border-width * 2))
```
Comment (jgarber623-gov) [inline src/_vendor/assets/stylesheets/uswds/elements/_inputs.scss]: Reverting to the original and will try and solve the deprecation warning down the line…

File: src/moving-allowances/determine-your-allowance/index.html [hunk 4]
```diff
@@ -6,7 +6,31 @@
...
+  <select id="move-type">
+    <option value=""></option>
+    <option value="CONUS">Within the Continental United States (CONUS)</option>
+    <option value="OCONUS">Outside Continental United States (OCONUS)</option>
+  </select>
+
+  <button>View Results</button>
+</form>
+
+<section class="usa-section moving-allowances-section" id="moving-allowances" tabindex="-1">
```
Comment (jgarber623-gov) [inline src/moving-allowances/determine-your-allowance/index.html]: Adding `tabindex` to make this element [focusable with JS](https://snook.ca/archives/accessibility_and_usability/elements_focusable_with_tabindex).

File: src/moving-allowances/determine-your-allowance/index.html [hunk 5]
```diff
@@ -6,7 +6,31 @@
...
+    <option value=""></option>
+    {% for pay_grade in site.data.pay_grades %}
+      <option value="{{ pay_grade.rank }}">{{ pay_grade.rank }}</option>
+    {% endfor %}
+  </select>
+
+  <label for="dependency-status">2. Select your <b>dependency status</b>.</label>
+  <select id="dependency-status">
+    <option value=""></option>
+  </select>
```
Comment (jgarber623-gov) [inline src/moving-allowances/determine-your-allowance/index.html]: @lfantano What is the wording for the `<option>`s for this `<select>`?

**Comment by lfantano** [comment]
Hey Jason-

I don't mind the yellow addition on the wide view. I'm debating whether it
should be a color other than yellow but need to think on that for a bit.

My only thought is that we should remove the color on mobile though so that
the side padding goes away-- maximizing the width we can display text on.

**Comment by jgarber623-gov** [comment]
Thanks, @lfantano! I'll make that viewport width-related change.

**Pull request merged**